api/pkg/setup/setup.go

```diff
@@ -17,8 +17,10 @@ func init() {
 }
 
 type ApiConfiguration struct {
-	Port     uint   `mapstructure:"port"`
-	LogLevel string `mapstructure:"log_level"`
+	Port      uint   `mapstructure:"port"`
+	LogLevel  string `mapstructure:"log_level"`
+	IssuerURL string `mapstructure:"oidc_issuer_url"`
```

Comment:
open to ideas on how to improve this configuration. I imagine we don't want to hardcode the client IDs into a config file and commit it to source code. Can we use viper to pull environment variables @hspitzley-czi ?

Reply:
You can do something like this: https://github.com/chanzuckerberg/happy/pull/634/files

Then set the values in `HAPPY_API_API_OIDC_CLIENT_ID` and `HAPPY_API_API_OIDC_ISSUER_URL` (note `API` appears twice, first because of the `happy_api` prefix and then again because the values are nested in the API config struct)

Reply:
so to deploy the API, we would need to have a configuration file and a set of environment variables? Or are you suggesting we use environment variables to fill in the values of the config file.

Reply:
If you merge the change in my PR we'll only need to set the values in `HAPPY_API_API_OIDC_CLIENT_ID` and  `HAPPY_API_API_OIDC_ISSUER_URL` env vars, viper will pick them up from there

Reply:
for reference, this is how the eng-portal config looks: https://github.com/chanzuckerberg/eng-portal/blob/0137f0f833c1caa33e9cfc3469ebf644bf9007c7/app-config.yaml#L74. The `${}` notation is used to add environment variables to the configuration file. What do you think of that?

Reply:
I tried briefly to do that with Viper and it didn't work, if you see a way to make that work I do think it would be preferred